crates/deltalake-core/src/logstore/mod.rs

```diff
@@ -0,0 +1,340 @@
+///   `get_latest_version` must return a version >= `v`, i.e. the underlying file system entry must
+///   become visible immediately.
+#[async_trait::async_trait]
+pub trait LogStore: Sync + Send {
+    /// Read data for commit entry with the given version.
+    /// TODO: return the actual commit data, i.e. Vec<Action>, instead?
+    async fn read_commit_entry(&self, version: i64) -> DeltaResult<Bytes>;
+
+    /// Write list of actions as delta commit entry for given version.
+    ///
+    /// This operation can be retried with a higher version in case the write
+    /// fails with [`TransactionError::VersionAlreadyExists`].
+    async fn write_commit_entry(
+        &self,
+        version: i64,
+        tmp_commit: &Path,
+    ) -> Result<(), TransactionError>;
+
+    /// Find latest version currently stored in the delta log.
+    async fn get_latest_version(&self, start_version: i64) -> DeltaResult<i64>;
+
+    /// Get underlying object store.
+    fn object_store(&self) -> Arc<dyn ObjectStore>;
+
+    /// [Path] to Delta log
+    fn to_uri(&self, location: &Path) -> String;
+
+    /// Get fully qualified uri for table root
+    fn root_uri(&self) -> String {
+        self.to_uri(&Path::from(""))
+    }
+
+    /// [Path] to Delta log
+    fn log_path(&self) -> &Path {
+        &DELTA_LOG_PATH
+    }
+
+    /// Check if the location is a delta table location
+    async fn is_delta_table_location(&self) -> ObjectStoreResult<bool> {
+        // TODO We should really be using HEAD here, but this fails in windows tests
+        let object_store = self.object_store();
+        let mut stream = object_store.list(Some(self.log_path())).await?;
+        if let Some(res) = stream.next().await {
+            match res {
+                Ok(_) => Ok(true),
+                Err(ObjectStoreError::NotFound { .. }) => Ok(false),
+                Err(err) => Err(err),
+            }
+        } else {
+            Ok(false)
+        }
+    }
+
+    #[cfg(feature = "datafusion")]
+    /// Generate a unique enough url to identify the store in datafusion.
+    /// The DF object store registry only cares about the scheme and the host of the url for
+    /// registering/fetching. In our case the scheme is hard-coded to "delta-rs", so to get a unique
+    /// host we convert the location from this `LogStore` to a valid name, combining the
+    /// original scheme, host and path with invalid characters replaced.
+    fn object_store_url(&self) -> ObjectStoreUrl;
+
+    /// Deletes object by `paths`.
+    async fn delete_batch(&self, paths: &[Path]) -> ObjectStoreResult<()> {
+        let object_store = self.object_store();
+        for path in paths {
+            match object_store.delete(path).await {
+                Ok(_) => continue,
+                Err(ObjectStoreError::NotFound { .. }) => continue,
+                Err(e) => return Err(e),
+            }
+        }
+        Ok(())
+    }
```

Comment:
IIRC, this was just a stop gap implementation by now we should just forward to `ObjectStores`s `delete_stream` method, which should fall back to more or less this, if the underlying store does not support batch deletes.

Reply:
As far as I can tell, this is entirely unused. There are places (vacuum, checkpointing) that directly call `delete_stream`. Since this is a breaking change anyway, would it make sense to just drop the implementation? 

On another note, this is the worst-performing implementation possible ;), as it doesn't even emit concurrent single-delete operations, but goes one by one in a sequential loop.

Reply:
Yes, lets just drop it! 😆